diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1bafda311..0e8772112 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,6 +70,9 @@ MARK_AS_ADVANCED(CLEAR CMAKE_CXX_COMPILER)
 MARK_AS_ADVANCED(GLUT_cocoa_LIBRARY)
 
 SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/common/cmake ${CMAKE_MODULE_PATH})
+IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  SET(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR}/install CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
+ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
 
 INCLUDE(test)
 
diff --git a/common/cmake/ispc.cmake b/common/cmake/ispc.cmake
index ec9e435c7..bad69c449 100644
--- a/common/cmake/ispc.cmake
+++ b/common/cmake/ispc.cmake
@@ -26,7 +26,7 @@ ENDMACRO ()
 IF (EMBREE_ISPC_SUPPORT)
 
 # ISPC versions to look for, in decending order (newest first)
-SET(ISPC_VERSION_WORKING "1.9.1" "1.9.0" "1.8.3" "1.8.2")
+SET(ISPC_VERSION_WORKING "1.9.2" "1.9.1" "1.9.0" "1.8.3" "1.8.2")
 LIST(GET ISPC_VERSION_WORKING -1 ISPC_VERSION_REQUIRED)
 
 IF (NOT EMBREE_ISPC_EXECUTABLE)
@@ -46,6 +46,7 @@ IF (NOT EMBREE_ISPC_EXECUTABLE)
   FOREACH(ver ${ISPC_VERSION_WORKING})
     FOREACH(suffix ${ISPC_DIR_SUFFIX})
       LIST(APPEND ISPC_DIR_HINT ${PROJECT_SOURCE_DIR}/../ispc-v${ver}-${suffix})
+      LIST(APPEND ISPC_DIR_HINT ${PROJECT_SOURCE_DIR}/../ispc/ispc-v${ver}-${suffix})
     ENDFOREACH()
   ENDFOREACH()
 
