diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2e9c856c..b096cd92 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -41,6 +41,11 @@ SET(OSPCOMMON_DIR ${COMPONENTS_DIR}/ospcommon)
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)
 SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${OSPCOMMON_DIR}/cmake)
 
+IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  SET(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR}/install CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
+ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+
+
 ##############################################################
 # CMake modules and macro files
 ##############################################################
diff --git a/cmake/ospray_macros.cmake b/cmake/ospray_macros.cmake
index a020b76b..0c343e89 100644
--- a/cmake/ospray_macros.cmake
+++ b/cmake/ospray_macros.cmake
@@ -40,7 +40,11 @@ function(ospray_verify_embree_features)
 endfunction()
 
 macro(ospray_find_embree EMBREE_VERSION_REQUIRED)
-  find_package(embree ${EMBREE_VERSION_REQUIRED} QUIET)
+  find_package(embree ${EMBREE_VERSION_REQUIRED} 
+    PATHS ${PROJECT_SOURCE_DIR}/../embree/install/lib64/cmake
+          ${PROJECT_SOURCE_DIR}/../embree/install/lib/cmake
+          ${PROJECT_SOURCE_DIR}/../embree/build
+    QUIET)
   if(NOT DEFINED EMBREE_INCLUDE_DIRS)
     message(FATAL_ERROR
             "We did not find Embree installed on your system. OSPRay requires"
diff --git a/components/ospcommon/cmake/ispc.cmake b/components/ospcommon/cmake/ispc.cmake
index c797af5f..9774ed0d 100644
--- a/components/ospcommon/cmake/ispc.cmake
+++ b/components/ospcommon/cmake/ispc.cmake
@@ -35,6 +35,7 @@ IF (NOT ISPC_EXECUTABLE)
   FOREACH(ver ${ISPC_VERSION_WORKING})
     FOREACH(suffix ${ISPC_DIR_SUFFIX})
       LIST(APPEND ISPC_DIR_HINT ${PROJECT_SOURCE_DIR}/../ispc-v${ver}-${suffix})
+      LIST(APPEND ISPC_DIR_HINT ${PROJECT_SOURCE_DIR}/../ispc/ispc-v${ver}-${suffix})
     ENDFOREACH()
   ENDFOREACH()
 
