diff --git a/CMakeLists.txt b/CMakeLists.txt
index a04979d..6b55b7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,10 @@ project(threadpool11)
 
 set(CMAKE_VERBOSE_MAKEFILE on)
 
+IF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+  SET(CMAKE_INSTALL_PREFIX ${PROJECT_SOURCE_DIR}/install CACHE PATH "Install path prefix, prepended onto install directories." FORCE)
+ENDIF(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+
 # SET a default build type if none was specified
 if(NOT CMAKE_BUILD_TYPE)
   message(STATUS "Setting build type to 'Release' as none was specified.")
@@ -10,7 +14,8 @@ if(NOT CMAKE_BUILD_TYPE)
   # SET the possible values of build type for cmake-gui
 endif()
 
-find_package(Boost)
+# pnav: disable external boost
+#find_package(Boost)
 
 if(Boost_FOUND)
   include_directories(${Boost_INCLUDE_DIR})
diff --git a/threadpool11/src/pool.cpp b/threadpool11/src/pool.cpp
index 751e4c6..c64f2da 100644
--- a/threadpool11/src/pool.cpp
+++ b/threadpool11/src/pool.cpp
@@ -23,6 +23,7 @@ This file is part of threadpool11.
 #include <boost/lockfree/queue.hpp>
 
 #include <algorithm>
+#include <vector>
 
 namespace threadpool11 {
 
